src/profiling/meter_manager.cpp

```diff
@@ -0,0 +1,71 @@
+#include "meter_manager.hpp"
+
+namespace nest {
+namespace mc {
+namespace util {
+
+using nlohmann::json;
+
+meter_manager::meter_manager() {
+    // add time-measurement meter
+    meters_.emplace_back(new time_meter());
+
+    // add memory consumption meter
+    // TODO
+
+    // add energy consumption meter
+    // TODO
+};
+
+void meter_manager::checkpoint(std::string name) {
+    checkpoint_names_.push_back(std::move(name));
```

Comment:
The meters do need not know about checkpoint names; they simply take measurements when requested.
I don't know if it is a good or bad thing that names could be repeated. We would have to think about how such an infrastructure could be used.